crates/test-framework/src/spicetest/datasets/mod.rs

```diff
@@ -284,6 +288,16 @@ impl SpiceTest<Completed> {
         self.state.query_durations.values().flatten().copied().sum()
     }
 
+    pub fn get_average_total_queries_executed(&self) -> Result<u32> {
+        // total query count across all workers
+        let total_query_count = self.state.query_durations.values().flatten().count();
+
+        // average query count per worker
+        Ok(u32::try_from(
+            total_query_count / self.state.parallel_count,
```

Comment:
Can this be a percentile? Really want to avoid averages. It sets a bad precedent.